controllers/storageclaim_controller.go

```diff
@@ -380,6 +382,9 @@ func (r *StorageClaimReconciler) reconcilePhases() (reconcile.Result, error) {
 					csiClusterConfigEntry.CephFS.SubvolumeGroup = data["subvolumegroupname"]
 					// delete groupname from data as its not required in storageclass
 					delete(data, "subvolumegroupname")
+					csiClusterConfigEntry.CephFS.KernelMountOptions = data[kernelMountOptionsKey]
+					// delete kernelmountoptions from data as its not required in storageclass
+					delete(data, kernelMountOptionsKey)
```

Comment:
We should not piggyback information destination for one resource (csi configmap) as part of another resource (storage class) to send CSI config data. The two should not be coupled and the fact that you need to delete pieces of information from the incoming data proves you are using the wrong approach. 

Please send all of the CSI config information as its own resource (a configmap with a `config.json` key) and handle it in a separate switch case! (similar to other resources we are sending from the provider). As a rule of thumb, each switch case need to be able to act independently!

@leelavg please make sure @rohan47's contribution here aligns with the proper way we set to send information from provider to client.

Reply:
@nb-ohad the existing mechanism is also sending data as part of storageclass that corresponds to it and we pick the interested items to configmap and this is also following the same.